### PR TITLE
feat: transform facility names from uppercase to sentence case

### LINKED ISSUE #5: feat: transform facility names from uppercase to sentence case
### Affected area

src/components/FacilityCard.tsx

### Problem to solve

Currently, the facility names are all in uppercase, which makes it a little less intuitive to read. It is also uppercase due to fetching of raw data from our OneMap API, which has all the address information in uppercase.

Image

### Proposed solution

To transform the facility names (location building) from uppercase to sentence case i.e. instead of 313 SOMERSET --> 313 Somerset.

```ts
        <div className="flex-1">
          <div className="flex items-center gap-2">
            <h2>
              {
                location.building
                  ? location.building
                  : location.block
                    ? `${location.block} ${location.road}`
                    : location.address
              }
            </h2>
```

### Alternatives considered

_No response_

### Out of scope

Do not change the input of what is to be displayed, only change the formatting of it!

### Acceptance criteria

- [ ] Clear UX/UI behavior defined
- [ ] Edge cases considered
- [ ] Tests added/updated
- [ ] Documentation updated

### Priority

P3 - Low

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kexin-Wei**
# Summary

added two functions: a general uppercase to sentence case function and a special handling function to skip numbers or special characters for case transformation and trim space, e.g. only make alphabetic letters to sentence case once found

## Related Issues / Tickets

- Closes #5 

## Contributor info

- LinkedIn profile: [Kexin-Wei](https://www.linkedin.com/in/kexin-wei-611s96/)

## Type of change

- [ ] Fix (non-breaking change that fixes an issue)
- [x] Feature (non-breaking change that adds functionality)
- [ ] Refactor (no functional change)
- [ ] Chore/Docs/Build
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How to Test
edit `src/data/locations.json`  
```json
  {
    "building": "ION ORCHARD MALL",
    "block": 2,
    "road": "ORCHARD TURN",
    "address": "2 ORCHARD TURN, SINGAPORE 238801",
    "postalCode": "238801",
    "latitude": 1.3040,
    "longitude": 103.8318,
    "opensAt": "09:00:00",
    "closesAt": "19:00:00"
  },

```
and check in the frontend

```bash
npm install
npm run db:fresh:local
npm run dev
```

## Screenshots / Recordings

N.A.

## Checklist

- [x] I ran the app locally and verified the change
- [x] I added/updated tests as needed
- [x] I updated documentation (README, comments, or storybook) as needed
- [ ] I updated environment docs if env vars changed
- [x] No sensitive info (keys, tokens) is committed
- [ ] Database migration required (see “Database Migration” section)

## Breaking Changes

N.A.

## Database Migration

N.A.

## Deployment Notes

N.A.

## Environment Changes

N.A.

## Additional Context

N.A.

### CODE REVIEW

**Comment by victoria-lo** [comment]
Hi @Kexin-Wei ! Thanks for the PR! To make it easy for maintainers to review, please always ensure that PRs indicate the issue number it resolves :) We will have a maintainer reviewing soon!

File: src/utils/text.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Converts a string to sentence case.
+ * @param str the input string
+ * @returns the string in sentence case
+ */
+export const toSentenceCase = (str: string): string => {
```
Comment (eileenchua) [inline src/utils/text.ts]: is this function being used anywhere?

**Review by eileenchua — COMMENTED** [review]
@Kexin-Wei, thanks for making the changes and ensuring test coverage with the edge cases! Please refer to my other comments below, though I also want to clarify if we should be using sentence case for this. Reminder to provide screenshots of the changes so that it helps with the review.

@natashaannn, for this issue... do we want to transform the place names to sentence case? Title case would look better (e.g. Mandarin Gallery vs Mandarin gallery) since these are place names. There would also be instances like "ION Orchard" that require more handling, otherwise it would be "Ion Orchard"

image

File: src/utils/text.ts (hunk 1 above)
Comment (Kexin-Wei) [reply src/utils/text.ts]: forget to call it, sorry, updated in the new commit

File: src/utils/text.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { describe, expect, it } from 'vitest';
+import { transformToSentenceCase } from './text';
+
+describe('Text transformation utilities', () => {
+  describe('skipTrimToSentenceCase', () => {
```
Comment (natashaannn) [inline src/utils/text.test.ts]: Don't forget to rename the describe block here since we are not using `skipTrimToSentenceCase` anymore! Also since this is for testing, we shouldn't be using the syntax for function names, and instead be providing descriptive context. Tests should ideally be as human-readable as possible as they serve as documentation as well! So for the describe block you can do something like 'when text is not sentence case'. See [this testing best practice guide](https://alexop.dev/posts/frontend-testing-guide-10-essential-rules/) for info!

File: src/utils/text.test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      { input: 'A', expected: 'A' },
+      { input: 'b', expected: 'B' },
+      { input: '123 MAIN STREET', expected: '123 Main street' },
+      { input: 'mIXed CaSe StRiNg', expected: 'Mixed case string' },
+      { input: '   LEADING AND TRAILING SPACES   ', expected: 'Leading and trailing spaces' },
+      { input: 'SPECIAL CHARACTERS !@#$', expected: 'Special characters !@#$' },
+      { input: '313@ORCHARD', expected: '313@Orchard' },
+    ];
+
+    it('correctly transforms normal cases to sentence case', () => {
```
Comment (natashaannn) [inline src/utils/text.test.ts]: Best practice for testing is to use 'should' followed by a verb for naming unit tests! Correctly isn't needed as it is assumed. I.e. 'should transform uppercase phrase to sentence case'

File: src/utils/text.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+describe('Text transformation utilities', () => {
+  describe('skipTrimToSentenceCase', () => {
+    const normalCases = [
+      { input: 'ION ORCHARD MALL', expected: 'Ion orchard mall' },
+      { input: 'FAR EAST PLAZA', expected: 'Far east plaza' },
+      { input: 'MANDARIN GALLERY', expected: 'Mandarin gallery' },
+    ];
+
+    const edgeCases = [
```
Comment (natashaannn) [inline src/utils/text.test.ts]: Rather than grouping edgeCases in one, best practice is to name each case specifically as individual tests!

File: src/utils/text.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { describe, expect, it } from 'vitest';
+import { transformToSentenceCase } from './text';
+
+describe('Text transformation utilities', () => {
+  describe('skipTrimToSentenceCase', () => {
+    const normalCases = [
```
Comment (natashaannn) [inline src/utils/text.test.ts]: `normalCases` might be a misnomer here - `upperCases` would be more explicitly descriptive. Also best to move this into a unit test case specifically for 'should change uppercase phrase to sentence case'

File: src/utils/text.test.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      { input: '313@ORCHARD', expected: '313@Orchard' },
+    ];
+
+    it('correctly transforms normal cases to sentence case', () => {
+      normalCases.forEach(({ input, expected }) => {
+        expect(transformToSentenceCase(input)).toBe(expected);
+      });
+    });
+
+    it('correctly handles edge cases', () => {
```
Comment (natashaannn) [inline src/utils/text.test.ts]: Should remove this in preference for individual cases!

File: src/utils/text.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ * - trims whitespace
+ * - uppercases the first alphabetic character and lowercases following letters
+ * - preserves leading non-letter characters; returns trimmed string if no letters
+ *
+ * Note: internal acronyms/mixed-case words will be lowercased (e.g. "NASA" -> "Nasa").
+ */
+
+export const transformToSentenceCase = (str: string): string => {
+  if (!str) {
+    return str;
```
Comment (natashaannn) [inline src/utils/text.ts]: If str is an empty string or falsy, it returns str directly (which could be undefined or null if someone accidentally passes that).
TypeScript guarantees str is a string here, but the if (!str) check implies you're expecting potential misuse — so better to standardize the return type to "" instead of forwarding non-string falsy values.

**Review by natashaannn — CHANGES_REQUESTED** [review]
Hey Kexin, thanks so much for submitting your PR! I remember you're a relatively new coder too, so this is awesome work. This PR might seem simple but it covers important practices like testing - so I'm so glad you took this up with us!

I referenced this in one of my comments, but it'll be great to take a look at thie [best practice for testing guide](https://alexop.dev/posts/frontend-testing-guide-10-essential-rules/) for naming conventions! You also mentioned you had trouble with creating good names, so best practice rules like these help in coming up with them!

After seeing the (very nicely done) test cases you've come up with, I realized that the `transformToSentenceCase` function you've done here might be better used in the facility card description, while for the facility card names, a better function might be `capitalizeFirstAlphabeticChar`.

So 'ION ORCHARD' becomes 'Ion Orchard' instead of 'Ion orchard'. 

Don't delete your transformtosentencecase function here, but instead, add a new text util function and new tests to the same Text transformation utilities test block for a new util called `capitalizeFirstAlphabeticChar`! Thereafter you can make the change in FacilityCard, where `transformToSentenceCase` is used for the facility descripition and `capitalizeFirstAlphabeticChar` is used for the name. 

Great work!!

**Comment by Kexin-Wei** [comment]
thank you a lot for your kind word and detailed instruction given the test cases @natashaannn 
i have tested the new function `capitalizeFirstAlphabeticChar` by editing the `src/data/locations.json` with code
```json
 {
    "building": "ION ORCHARD MALL",
    "block": 2,
    "road": "ORCHARD TURN",
    "address": "2 ORCHARD TURN, SINGAPORE 238801",
    "postalCode": "238801",
    "latitude": 1.3040,
    "longitude": 103.8318,
    "opensAt": "09:00:00",
    "closesAt": "19:00:00"
  }
```

and  `src/data/facilities.json` with 
```json
[
  {
    "locationAddress": "2 Orchard Turn, Singapore 238801",
    "facilityTypeName": "Baby Room",
    "floor": "B1",
    "isAccessible": true,
    "description": "NEAR FEMALE RESTROOMS",
    "hasDiaperChangingStation": true,
    "hasLactationRoom": true
  },
  {
    "locationAddress": "333A Orchard Rd, Singapore 238897",
    "facilityTypeName": "Baby Room",
    "floor": "L2",
    "isAccessible": true,
    "description": "nEAR rESTROOM",
    "hasDiaperChangingStation": true,
    "hasLactationRoom": false
  },
  {
    "locationAddress": "14 Scotts Rd, Singapore 228213",
    "facilityTypeName": "Diaper Changing Station",
    "floor": "L4",
    "isAccessible": true,
    "description": "near escalator",
    "hasDiaperChangingStation": true,
    "hasLactationRoom": false
  }
]
```

the screenshot for the test result is
Screenshot from 2025-11-03 23-04-38
